docs/how_to_dev_frontend_en.md

```diff
@@ -0,0 +1,102 @@
+# VisualDL Web app Develop
+
+VisualDL has two three components.
+1. The Python/C++ SDK that logs the data during training.
+1. The single page client side app that visualized training data.
+1. The server(powered on Flask) that parses the saved data and delivers to the client side app for display.
+
+The server provided the needed data and pass to the client side to draw scalars or histograms. The users can visualize the training progress and make a better network architecture evaluation.
+
+This article will go over the basic web-app architecture and development steps.
+
+## npm package
+The VisualDL uses npm to manage the JavaScript files. npm is the package manager for JavaScript and is a great tool to manage multiple dependencies. To confirm that you have npm installed. Please run this command
+``` bash
+npm -v
+```
+If your environment does not have npm already, please [download npm](https://www.npmjs.com/get-npm).
+
+The basic packaging information is stored in `package.json`. It specifies build scripts and dependencies for each environment.
+npm reads `package.json` to build different package according to the packaging script.
+If you are not familiar with npm, please refer to the [npm documentation](https://docs.npmjs.com/) for more detail.
+
+Here is a list of the dependencies used in the VisualDL.
```

Comment:
This might be too detailed for a quick start development guide, and also the requirements/versions may change.  I think we should remove this section.